src/buildstream/element.py

```diff
@@ -1272,6 +1272,22 @@ def _get_cache_key(self, strength=_KeyStrength.STRONG):
         else:
             return self.__weak_cache_key
 
+    # _get_cas_digest():
+    #
+    # Returns the CAS digest
+    #
+    # Returns:
+    #    (str|None): The CAS digest for this Element, or None
+    #
+    # None is returned if no CAS artifact is available for this element.
+    #
+    def _get_cas_digest(self):
+        if self.__artifact is None:
+            return None
+        if not self.__artifact.query_cache():
```

Comment:
Is this still needed? Now that you're correctly querying the cache at the beginning?